fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -5067,19 +5067,9 @@ Standalone<VectorRef<KeyValueRef>> generateOldFormatMutations(
 	for (int part = 0; part * CLIENT_KNOBS->MUTATION_BLOCK_SIZE < param2Concat.size(); part++) {
 		KeyValueRef backupKV;
 		// Assign the second parameter as the part
-		backupKV.value = param2Concat.substr(part * CLIENT_KNOBS->MUTATION_BLOCK_SIZE,
-		                                     std::min(param2Concat.size() - part * CLIENT_KNOBS->MUTATION_BLOCK_SIZE,
-		                                              CLIENT_KNOBS->MUTATION_BLOCK_SIZE));
-		// Write the last part of the mutation to the serialization, if the buffer is not defined
-		if (!partBuffer) {
-			// part = 0
-			wrParam1 << bigEndian32(part);
-			partBuffer = (uint32_t*)((char*)wrParam1.getData() + wrParam1.getLength() - sizeof(uint32_t));
-		} else {
-			// part > 0
-			*partBuffer = bigEndian32(part);
-		}
-		backupKV.key = wrParam1.toValue();
+		Standalone<StringRef> backupVal = getBackupValue(param2Concat, part);
```

Comment:
it seems i have to create a standalone in the caller, if i have `backupKV.value = getBackupValue(param2Concat, part)` directly, then the standalone<stringref> returned will somehow be destructed, am i understanding it correctly? @jzhou77

Reply:
Not sure what you mean here. If you mean `backupKV.value = getBackupValue(param2Concat, part);`, then yes, the returned `Standalone<StringRef>` is destructed, since `backupKV.value` is a `StringRef`.

`getBackupValue()` is taking a substring from `param2Concat`, so a `StringRef` is good enough whenever `param2Concat` is valid, i.e., arena holds the memory. I see `backupKV` is only used below by `results`. So it's simple just change the `getBackupValue()` to return a `StringRef`.

Reply:
`getBackupValue()` returning a StringRef is desirable, since it avoids a copy.